ngraph/core/src/op/util/convert_color_i420_base.cpp

```diff
@@ -0,0 +1,216 @@
+    auto out_shape = shape_y;
+    auto out_type = y_type;
+    if (out_shape.rank().is_dynamic()) {
+        out_shape = PartialShape{Dimension::dynamic(), Dimension::dynamic(), Dimension::dynamic(), 3};
+    }
+    out_shape[i420_op::C_DIM] = 3;  // 3 is number of channels (R, G, B)
+    if (single_plane) {
+        if (shape_y.rank().is_static() && shape_y[i420_op::H_DIM].is_static()) {
+            NODE_VALIDATION_CHECK(this,
+                                  shape_y[i420_op::H_DIM].get_length() % 3 == 0,
+                                  "I420 image height shall be divisible by 3, but it is ",
+                                  shape_y[i420_op::H_DIM].get_length());
+            // E.g. if input shape height is 720 for I420, then real image height is 720 * 2 / 3 = 480
+            out_shape[i420_op::H_DIM] = shape_y[i420_op::H_DIM].get_length() * 2 / 3;
+        }
+    } else {
+        auto u_type = get_input_element_type(1);
+        auto v_type = get_input_element_type(2);
+        NODE_VALIDATION_CHECK(this,
+                              ov::element::Type::merge(out_type, out_type, u_type),
+                              "Y, U, V inputs shall have compatible types, got ",
+                              y_type,
+                              u_type,
+                              v_type);
+        NODE_VALIDATION_CHECK(this,
+                              ov::element::Type::merge(out_type, out_type, v_type),
+                              "Y, U, V inputs shall have compatible types, got ",
+                              y_type,
+                              u_type,
+                              v_type);
+        // Validate Y/U/V shapes compatibility
+        const auto& shape_u = get_input_partial_shape(1);
+        NODE_VALIDATION_CHECK(this,
+                              shape_u.compatible(one_channel_nhwc_shape),
+                              "U input shall have 4 dimensions (N, H, W, C) with channels dimension equal to 1, got ",
+                              shape_u);
+        const auto& shape_v = get_input_partial_shape(2);
+        NODE_VALIDATION_CHECK(this,
+                              shape_v.compatible(one_channel_nhwc_shape),
+                              "V input shall have 4 dimensions (N, H, W, C) with channels dimension equal to 1, got ",
+                              shape_v);
+        NODE_VALIDATION_CHECK(this,
+                              shape_u.compatible(shape_v),
+                              "U shape shall be compatible with V shape: ",
+                              shape_u,
+                              shape_v);
+        auto shape_uv = shape_u;
+        PartialShape::merge_into(shape_uv, shape_v);
+        if (shape_uv.rank().is_static()) {
+            if (!shape_uv[i420_op::H_DIM].is_dynamic()) {
+                shape_uv[i420_op::H_DIM] *= 2;
+            }
+            if (!shape_uv[i420_op::W_DIM].is_dynamic()) {
+                shape_uv[i420_op::W_DIM] *= 2;
+            }
```

Comment:
Basically we can multiply dimensions without checking if it is dynamic. Please [see](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/dimension.cpp#L38) 
```suggestion
            shape_uv[i420_op::H_DIM] *= 2;
            shape_uv[i420_op::W_DIM] *= 2;
```

Feel free to change it in the next PR -- as you stated in the description there will be a follow up PR.